telegram/ext/filters.py

```diff
@@ -179,7 +170,7 @@ class MessageFilter(BaseFilter):
 
     __slots__ = ()
 
-    def __call__(self, update: Update) -> Optional[Union[bool, DataDict]]:
+    def check_update(self, update: Update) -> Optional[Union[bool, DataDict]]:
```

Comment:
actually we can't, since `filters.UpdateType.ALL` does not include messages from `update.callback_query`, whereas `update.effective_message` will include the message update from that, making `MessageHandler` handling updates from callback queries.

Reply:
You're right 👍 I still think that `MessageFilter.check_update` should check if `update.effective_message is None`.

One could even argue that it should check if the update is an actual message, i.e. if `update.(edited_)message/channel_post` is present … I'm not sure. Guess I'm okay with both. If we keep it with `Filters.Update.ALL`, I'd like to have comments added to `Message/CommandHandler` to avoid future confusion on this :)

Reply:
applied the changes